commons/pom.xml

```diff
@@ -17,6 +17,7 @@
       <dependency>
          <groupId>com.github.ben-manes.caffeine</groupId>
          <artifactId>caffeine</artifactId>
+         <optional>true</optional>
```

Comment:
ic. Are there many use cases to have `infinispan-commons`, yet no have caffeine then?
If all (or most) uses will need this, one might as well keep it as a non-optional dependency.